sdk/src/main/java/org/coralibre/android/sdk/internal/device_info/DeviceList.java

```diff
@@ -0,0 +1,184 @@
+package org.coralibre.android.sdk.internal.device_info;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import android.os.Build;
+import android.preference.PreferenceManager;
+import android.util.Log;
+
+import org.coralibre.android.sdk.BuildConfig;
+import org.coralibre.android.sdk.R;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.coralibre.android.sdk.internal.device_info.ConfidenceLevel.LOW;
+import static org.coralibre.android.sdk.internal.device_info.ConfidenceLevel.NONE;
+
+public class DeviceList {
+    private static final String TAG = DeviceInfo.class.toString();
+
+    public static final String DEVICE_INFO_KEY = "DeviceList_device_info_key";
+
+    private static final int LIST_FILE_ID = R.raw.en_calibration_2020_08_12;
+    private static final int LIST_LINE_COUNT = 11810;
+    public static final DeviceInfo DEFAULT_INFO = new DeviceInfo(
+            "unknown", "unknown", "unknown", 4, -25, NONE);
+
+    final DeviceInfo[] deviceList;
+    final SharedPreferences sharedPreferences;
+
+    public class FindingsResult {
+        public final List<DeviceInfo> manufacturerFindings;
+        public final List<DeviceInfo> oemDeviceFindings;
+        public final List<DeviceInfo> oemModelFindings;
+
+        public FindingsResult(List<DeviceInfo> manufacturerFindings,
+                              List<DeviceInfo> oemDeviceFindings,
+                              List<DeviceInfo> oemModelFindings) {
+            this.manufacturerFindings = Collections.unmodifiableList(manufacturerFindings);
+            this.oemDeviceFindings = Collections.unmodifiableList(oemDeviceFindings);
+            this.oemModelFindings = Collections.unmodifiableList(oemModelFindings);
+        }
+    }
+
+    public DeviceList(Context context, SharedPreferences sharedPrefs) throws IOException {
+        deviceList = loadDeviceInfoTable(context);
+        sharedPreferences = sharedPrefs;
+    }
+
+    public static DeviceInfo[] loadDeviceInfoTable(Context context) throws IOException {
+        BufferedReader infoCVS = new BufferedReader(new InputStreamReader(
+                context.getResources().openRawResource(LIST_FILE_ID)));
+        DeviceInfo[] deviceList = new DeviceInfo[LIST_LINE_COUNT - 1];
+        String line = infoCVS.readLine(); // discard the first line as it just contains the header
+        int i = 0;
+        while ((line = infoCVS.readLine()) != null) {
+            deviceList[i] = new DeviceInfo(line);
+            i++;
+        }
+        infoCVS.close();
+        return deviceList;
+    }
+
+    public DeviceInfo getInfo(final String manufacturer, final String device, final String model) {
```

Comment:
Maybe rename this or add a comment, to make it more clear that this tries to do an exact matching

Reply:
done